lib/facility.rb

```diff
@@ -1,14 +1,50 @@
+
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees,
+              :plate_type
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @plate_type = :regular
```

Comment:
This plate_type attribute is not an attribute of the Facility, but of the Vehicle. So we don't want this instance variable here, we'd want it in the Vehicle class.